aptos-move/block-executor/src/view.rs

```diff
@@ -1534,6 +1541,21 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TModuleView
             state_key,
         );
 
+        // Enforce feature gating V2 loader implementation: TModuleView is no longer used in
+        // V2 interfaces because we implement storage traits directly. Use a debug assert to
+        // panic in tests, adn invariant violation for non-debug builds.
+        if self.runtime_environment.vm_config().use_loader_v2 {
```

Comment:
if we only need use_loader_v2, then can we pass that to view constructor (instead of the whole runtime environment)?

Reply:
We need the environment to run verification and deserialization, it is mostly used in `modules_and_scripts.rs`